### ISSUE TITLE
Hide or change ray length, when using tool that creates content at specific offset/distance.

### ISSUE THREAD

**Issue by dunity** [Unity-Technologies/EditorVR#270]
When using a tool that creates/places content in the scene (consider the create cube tool), the ray length should represent the world position (offset from input device) at which the content will created.  In other cases, with tools whose interaction doesn't result in the ray having any applicable value, the ray should be hidden.  When moving back to a tool in which the ray visuals do apply to interaction (the select tool, or blink, etc), the ray should then again be shown.

**Comment by stella3d** [comment]
makes sense to me.  Should we make the ray length user-settable, you think, or just let the tool set it ?

**Comment by mtschoen** [comment]
This could be something we add to the "visibility settings" I set up in my Menu Hide PR.

**Comment by stella3d** [comment]
Menu Hide PR, the one PR to solve all problems

**Comment by dunity** [comment]
@stella3d I'd go with allowing the Tools to define the length.  Some tools will need to change this length dynamically, or at specific times/states.
